patches/server/0994-Add-Adventure-components-title-and-originalTitle-in-.patch

```diff
@@ -0,0 +1,157 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: LeonidM <wildhs.botm@gmail.com>
+Date: Fri, 4 Aug 2023 16:05:08 +0300
+Subject: [PATCH] Add Adventure components title and originalTitle in
+ InventoryView
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
+index 081e6c7ce4a0185ba75e48bbb581fd531cdb10a8..7d61c16ea889f83e84d80d37fcc5dc16ace6de15 100644
+--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
++++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
+@@ -50,6 +50,10 @@ public class CraftContainer extends AbstractContainerMenu {
+         this(new InventoryView() {
+ 
+             private final String originalTitle = inventory instanceof CraftInventoryCustom ? ((CraftInventoryCustom) inventory).getTitle() : inventory.getType().getDefaultTitle(); // Paper
++            // Paper start
++            private final net.kyori.adventure.text.Component originalTitleComponent = inventory instanceof CraftInventoryCustom ? ((CraftInventoryCustom) inventory).title() : inventory.getType().defaultTitle(); // Paper
++            private net.kyori.adventure.text.Component titleComponent = this.originalTitleComponent;
++            // Paper end
+             private String title = this.originalTitle;
+ 
+             @Override
+@@ -75,24 +79,39 @@ public class CraftContainer extends AbstractContainerMenu {
+             // Paper start
+             @Override
+             public net.kyori.adventure.text.Component title() {
+-                return inventory instanceof CraftInventoryCustom custom ? custom.title() : inventory.getType().defaultTitle(); // Paper
++                return this.titleComponent;
++            }
++
++            public void title(net.kyori.adventure.text.Component title) {
++                CraftInventoryView.sendInventoryTitleChange(this, title);
++                this.titleComponent = title;
++                this.title = net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().serialize(this.titleComponent);
++            }
++
++            @Override
++            public net.kyori.adventure.text.Component originalTitle() {
++                return this.originalTitleComponent;
+             }
+             // Paper end
+ 
+             @Override
++            @Deprecated // Paper
```

Comment:
No need to annotate internal methods for deprecation.
(Repeats in a few spaces in the patch)